### DISCUSSION TITLE
Translation components no longer visible.

### DISCUSSION THREAD

**Discussion by chimosky** [WeblateOrg/weblate#10499] [Q&A]
After upgrading the postgres container recently, following the [upgrade](https://docs.weblate.org/en/latest/admin/install/docker.html#upgrading-the-docker-container) instructions, I can't see any of the previously added translation componet for some reason and each translation project has a padlock right beside it.

![Screenshot from 2023-11-28 17-28-17](https://github.com/WeblateOrg/weblate/assets/16203834/d49ada7c-5b18-4ed4-8fcb-0b35dddd1d76)

The added translation components can be seen in the container but just not visible for some reason.

Please help.

**Comment by nijel** [comment]
The padlock indicates that there are locked components in a project.

> The added translation components can be seen in the container but just not visible for some reason.

Where they can be seen and where not?

**Reply by chimosky** [reply]
> The padlock indicates that there are locked components in a project.

How do I unlock these components?

> Where they can be seen and where not?

They can be seen in the container `/app/data/vcs/$projectname` and they can't be seen from the web interface even with admin access.

**Reply by nijel** [reply]
Ah, there is a bug in the padlock indication that it is shown for blank projects as well. I've just fixed that in ebe8fcd892.

When the components are not visible in the web interface, then something went wrong with the database upgrade. Did it show any errors? Apparently, only some tables were transferred.

**Reply by chimosky** [reply]
> When the components are not visible in the web interface, then something went wrong with the database upgrade. Did it show any errors? Apparently, only some tables were transferred.

I can't remember clearly but I think it first showed an error about using the weblate db and then it eventually was able to use it.

I can delete and restore the db again as the backup is still there, and show you the output.

**Reply by nijel** [reply]
It really looks like the restore (or backup) is incomplete and contains only part of the data.

**Reply by chimosky** [reply]
If that's the case then it really sucks as the backup was done with the command on the installation page and it being incomplete is something I couldn't foresee, I'm guessing this means that I'll need to get rid of the weblate volume and start afresh or is there anything that can be done to salvage the situation?

**Reply by chimosky** [reply]
@nijel how do I unlock a project, even after starting afresh and creating new projects, they all seem to be locked.

**Reply by nijel** [reply]
The empty project is shown as locked, but it is not, see https://github.com/WeblateOrg/weblate/discussions/10499#discussioncomment-7705405

**Reply by nijel** [reply]
As for the incompleteness, you should check for exit codes and error messages, most likely it didn't happen without a reason...

**Reply by chimosky** [reply]
Yeah I noticed this after I uploaded a component, thanks!

**Comment by chimosky** [comment]
> As for the incompleteness, you should check for exit codes and error messages, most likely it didn't happen without a reason...

True, when I'd backed up the db there weren't any errors so the issue might not be from that. Thanks!
